Mathlib/Algebra/Colimit/ModuleRing.lean

```diff
@@ -237,85 +227,36 @@ end functorial
 
 end Basic
 
-section Totalize
+section equiv
 
-open Classical in
-/-- `totalize G f i j` is a linear map from `G i` to `G j`, for *every* `i` and `j`.
-If `i ≤ j`, then it is the map `f i j` that comes with the directed system `G`,
-and otherwise it is the zero map. -/
-noncomputable def totalize (i j) : G i →ₗ[R] G j :=
-  if h : i ≤ j then f i j h else 0
+variable [Nonempty ι] [IsDirected ι (· ≤ ·)] [DecidableEq ι] [DirectedSystem G (f · · ·)]
+open _root_.DirectLimit
 
-variable {G f}
+/-- The direct limit constructed as a quotient of the direct sum is isomorphic to
+the direct limit constructed as a quotient of the disjoint union. -/
+def linearEquiv : DirectLimit G f ≃ₗ[R] _root_.DirectLimit G f :=
```

Comment:
So now we have two definitions of `DirectLimit`, whereas we had only one in the past, right?
One is a bit more general, but the other has better defeqs and is easier to work with. Do I understand that correctly?

I think part of your PR commit message contains useful commentary that should also be embedded in a module docstring in one of these files.

Reply:
There were [Module.DirectLimit](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/DirectLimit.html#Module.DirectLimit), [AddCommGroup.DirectLimit](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/DirectLimit.html#AddCommGroup.DirectLimit) and [Ring.DirectLimit](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/DirectLimit.html#Ring.DirectLimit), and I'm now adding DirectLimit in the root space. The three existing `DirectLimit`s are in fact constructions of general colimits in the respective categories, while mine is restricted to directed colimits but applies to all kinds of algebraic structures. The point is that when the preorder is directed, the underlying type of all DirectLimits can be taken to be the same, no matter what algebraic structure you consider; in fact I simply take the DirectLimit in Type* and put all kinds of algebraic structures on it. The equivalence relation on that defines the DirectLimit in Type* is easy to understand, and is used to golf the `of.zero_exact` lemmas.

I'll see what part of the PR message is appropriate to include in the module docstrings.

Reply:
Thanks, this message is also very useful. So please also add a version of that info to the module docstring.

Reply:
Please check [2122ea8](https://github.com/leanprover-community/mathlib4/pull/19893/commits/2122ea8df7114d975d6bd6562e2b09032afb75dc); is there anything I should add?